### PR TITLE
Tutorial on updating tutorials

### PR DESCRIPTION
**Pull request by nomadscientist**
I constantly work with SPOC to get tutorials updated, which is not easy to do. I end up onboarding new people over and over again, and trying to centralise and update that list and then repeat it over and over is a pain. Plus, we want to make these set-ups easier to use during Co-Fests.

Therefore, I've made a tutorial on how I go about updating a tutorial. I would be DELIGHTED to hear if people have ideas on how to streamline this, or other links that would help. Let's make it as easy as possible to contribute to the GTN sustainability!

### CODE REVIEW

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+{% include _includes/cyoa-choices.html option1="No" option2="Yes" default="No"
+       text="Do you already have a tutorial and workflow you want to update?" %}
+
+<div class="No" markdown="1">
+
+> <hands-on-title>Check for an outdated workflow</hands-on-title>
+>
+> 1. **Find a {% icon hands_on %} Hands-on tutorial** in your training topic of interest (for example, Single-cell)
+> 2. **Select the {% icon workflow %} workflow** from the header
+> 3. **Import the workflow** to your server
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:50]:
```suggestion
> 3. **Import the workflow** to your favorite Galaxy server
```

Comment (bebatut) [reply topics/contributing/tutorials/updating_tutorial/tutorial.md:50]: Maybe we could add a snipper for it

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+# Phase 1: Find a tutorial with an outdated workflow
+
+{% include _includes/cyoa-choices.html option1="No" option2="Yes" default="No"
+       text="Do you already have a tutorial and workflow you want to update?" %}
+
+<div class="No" markdown="1">
+
+> <hands-on-title>Check for an outdated workflow</hands-on-title>
+>
+> 1. **Find a {% icon hands_on %} Hands-on tutorial** in your training topic of interest (for example, Single-cell)
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:48]: Have you tried to put the `div` directly in the hands-on box? As the steps are pretty similar except the first one, that could be great to have only 1 hands-on box with 2 different step 1. That will avoid the redundancy

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+</div>
+
+# Phase 2: Check that nobody else is working on this
+
+It's always a good idea to check, just in case!
+
+You can:
+ - Send a message to the [GTN Matrix Channel](https://matrix.to/#/#Galaxy-Training-Network_Lobby:gitter.im) is your quickest way forward.
+ - Search through the [GTN Github Repository](https://github.com/galaxyproject/training-material) for existing draft Pull Requests.
+ - Check with [individual communities](https://galaxyproject.org/community/sig/), who may have their own method of tracking. For example, the [🖖🏾Single-cell & sPatial Omics Community](https://galaxyproject.org/community/sig/singlecell/) have a shared [Click-Up board](https://sharing.clickup.com/9015477668/b/h/5-90152810734-2/557452707486fef) at the time of writing.
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:78]:
```suggestion
 - Check with [individual communities](https://galaxyproject.org/community/sig/), who may have their own method of tracking. 
 
      For example, the [🖖🏾Single-cell & sPatial Omics Community](https://galaxyproject.org/community/sig/singlecell/) have a shared [Click-Up board](https://sharing.clickup.com/9015477668/b/h/5-90152810734-2/557452707486fef) at the time of writing.
```

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+# Phase 3: Update the workflow
+
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:81]: Could you add a sentence explaining that updating the workflow means using the last tool version? Thanks 😄

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+>
+> 1. Select {% icon galaxy-wf-options %} **Workflow Options** from the **Workflow Editor**
+> 2. Select {% icon upgrade_workflow %} **Upgrade workflow** to automatically update the tools.
+> 3. Address any issues that may arise.
+> 4. **Save** the workflow.
+{: .hands_on}
+
+
+# Phase 4: Test & fix the workflow
+
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:92]: Could you add a sentence explaining why testing is important? Thanks 😄

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+>    {% snippet faqs/galaxy/histories_sharing.md %}
+>
+{: .hands_on}
+
+# Phase 5: Update the tutorial
+
+> <warning-title>If you run out of time</warning-title>
+> You have already completed a large chunk of work (updated workflow + answer key history) to get here, and we don't want to lose it!
+> Updating the tutorial to match the updated workflow can be a separate contribution. So if you get to this point and run out of time, please:
+> 1. Create an **issue** on the GTN Github Repo and include shareable links to your updated workflow & answer key history
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:116]:
```suggestion
> 1. Create an **issue** on the GTN Github repositoty and include shareable links to your updated workflow & answer key history
```

Comment (bebatut) [reply topics/contributing/tutorials/updating_tutorial/tutorial.md:116]: I am not sure to understand what is "answer key history"

File: topics/contributing/tutorials/updating_tutorial/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+├── README.md
+├── metadata.yaml
+├── images
+├── docker
+│   ├── Dockerfile
+├── slides
+│   ├── index.html
+├── tutorials
+│   ├── tutorial1
+│   │   ├── tutorial.md
```
Comment (bebatut) [inline topics/contributing/tutorials/updating_tutorial/tutorial.md:135]:
```suggestion
│   │   ├── tutorial.md
│   │   ├── images
```

**Review by bebatut — COMMENTED** [review]
Added some suggestions and comments.
Thanks a lot @nomadscientist for this nice tutorial.

File: topics/contributing/tutorials/updating_tutorial/tutorial.md (hunk 2 above)
Comment (nomadscientist) [reply topics/contributing/tutorials/updating_tutorial/tutorial.md:48]: I'll try!

File: topics/contributing/tutorials/updating_tutorial/tutorial.md (hunk 6 above)
Comment (nomadscientist) [reply topics/contributing/tutorials/updating_tutorial/tutorial.md:127]: added in labels with `` throughout to emphasise this

**Comment by nomadscientist** [comment]
Thanks for your excellent review, I have addressed all comments :)

File: topics/contributing/tutorials/updating_tutorial/tutorial.md (hunk 2 above)
Comment (bebatut) [reply topics/contributing/tutorials/updating_tutorial/tutorial.md:49]: Damn it

**Review by bebatut — APPROVED** [review]

**Pull request merged**